File: do_shelf_update.js
const fs = require('fs');
let content = fs.readFileSync('mobile-app/app/journal.tsx', 'utf8');
content = content.replace(/\r\n/g, '\n');

const oldBottom = `<View style={[styles.bottomSection, compact && styles.bottomSectionCompact]}>
          <View style={[styles.actionCard, compact && styles.actionCardCompact]}>
            <View style={styles.actionCardContentRow}>
              <View style={styles.actionCardTextCol}>
                <Text style={[styles.actionCardTitle, compact && styles.actionCardTitleCompact]}>Express Yourself</Text>
                <Text style={[styles.actionCardSubtitle, compact && styles.actionCardSubtitleCompact]}>How are you feeling today?</Text>
              </View>
              <Image source={BOOK_IMAGE} style={[styles.actionBookImage, compact && styles.actionBookImageCompact]} resizeMode="contain" />
            </View>

            <View style={styles.actionButtonsRow}>
              <Pressable
                style={[styles.primaryActionButton, compact && styles.primaryActionButtonCompact]}
                onPress={() => router.push("/write-entry?mode=new")}
              >
                <Ionicons name="pencil" size={compact ? 18 : 20} color="#FFFFFF" style={{ marginRight: 6 }} />
                <Text style={[styles.primaryActionText, compact && styles.primaryActionTextCompact]}>Add Entry</Text>
              </Pressable>

              <Pressable
                style={[styles.secondaryActionButton, compact && styles.secondaryActionButtonCompact]}
                onPress={() => router.push("/journal-entries")}
              >
                <Ionicons name="book-outline" size={compact ? 18 : 20} color="#4D6558" style={{ marginRight: 6 }} />
                <Text style={[styles.secondaryActionText, compact && styles.secondaryActionTextCompact]}>Entries</Text>
              </Pressable>
            </View>
          </View>
        </View>`;

const newBottom = `<View style={[styles.bottomSection, compact && styles.bottomSectionCompact]}>
          <View style={styles.shelfHeaderRow}>
            <Text style={[styles.shelfTitle, compact && styles.shelfTitleCompact]}>Your Journals</Text>
            <Pressable style={styles.viewAllButton} onPress={() => router.push("/journal-entries")}>
              <Text style={styles.viewAllText}>View All</Text>
              <Ionicons name="chevron-forward" size={16} color="#6B7B8C" />
            </Pressable>
          </View>

          <View style={styles.shelfContainer}>
            {/* Shelf Book 1 */}
            <Pressable style={styles.shelfBookItem} onPress={() => router.push("/write-entry?mode=new-muni")}>
              <View style={[styles.bookWrapper, compact && styles.bookWrapperCompact]}>
                <Image source={BOOK_IMAGE} style={[styles.bookBaseImage, compact && styles.bookBaseImageCompact]} resizeMode="contain" />
                <View style={[styles.bookDeco, styles.bookDecoMuni]}>
                  <Ionicons name="chatbubbles" size={20} color="#FFFFFF" />
                </View>
              </View>
              <Text style={styles.bookLabel}>Guided Journal</Text>
              <Text style={styles.bookSubLabel}>With Muni</Text>
            </Pressable>

            {/* Shelf Book 2 */}
            <Pressable style={styles.shelfBookItem} onPress={() => router.push("/write-entry?mode=new-solo")}>
              <View style={[styles.bookWrapper, compact && styles.bookWrapperCompact]}>
                <Image source={BOOK_IMAGE} style={[styles.bookBaseImage, compact && styles.bookBaseImageCompact]} resizeMode="contain" />
                <View style={[styles.bookDeco, styles.bookDecoSolo]}>
                  <Ionicons name="pencil" size={20} color="#FFFFFF" />
                </View>
              </View>
              <Text style={styles.bookLabel}>Solo Journal</Text>
              <Text style={styles.bookSubLabel}>Free write</Text>
            </Pressable>
          </View>
          <View style={styles.shelfBoard} />
        </View>`;

if (content.includes(oldBottom)) {
  content = content.replace(oldBottom, newBottom);
  console.log("Replaced Bottom!");
} else {
  console.log("Could not find Bottom");
}

const styleStart = content.indexOf('  actionCard: {');
const styleEnd = content.indexOf('  modalBackdrop: {');
if (styleStart !== -1 && styleEnd !== -1) {
    const newStyles = `  shelfHeaderRow: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between",
    marginBottom: 16,
    paddingHorizontal: 4 },
  shelfTitle: {
    color: "#34475A",
    fontSize: 20,
    lineHeight: 26,
    fontFamily: "Outfit-Bold" },
  shelfTitleCompact: {
    fontSize: 18,
    lineHeight: 24 },
  viewAllButton: {
    flexDirection: "row",
    alignItems: "center",
    columnGap: 2 },
  viewAllText: {
    color: "#6B7B8C",
    fontSize: 14,
    lineHeight: 18,
    fontFamily: "Outfit-Bold" },
  shelfContainer: {
    flexDirection: "row",
    justifyContent: "space-evenly",
    alignItems: "flex-end",
    paddingHorizontal: 10,
    zIndex: 2,
    marginBottom: -4 },
  shelfBookItem: {
    alignItems: "center",
    width: "45%" },
  bookWrapper: {
    position: "relative",
    width: 120,
    height: 140,
    alignItems: "center",
    justifyContent: "center",
    marginBottom: 8 },
  bookWrapperCompact: {
    width: 100,
    height: 120 },
  bookBaseImage: {
    width: "100%",
    height: "100%" },
  bookBaseImageCompact: {
    width: "90%",
    height: "90%" },
  bookDeco: {
    position: "absolute",
    width: 36,
    height: 36,
    borderRadius: 18,
    alignItems: "center",
    justifyContent: "center",
    borderWidth: 2,
    borderColor: "#FFFFFF",
    shadowColor: "#000",
    shadowOpacity: 0.1,
    shadowRadius: 4,
    shadowOffset: { width: 0, height: 2 },
    elevation: 3 },
  bookDecoMuni: {
    backgroundColor: "#7BCB45",
    bottom: 15,
    right: 5 },
  bookDecoSolo: {
    backgroundColor: "#FFA726",
    bottom: 15,
    left: 5 },
  bookLabel: {
    color: "#34475A",
    fontSize: 15,
    fontFamily: "Outfit-Bold",
    marginBottom: 2 },
  bookSubLabel: {
    color: "#7B8D74",
    fontSize: 12 },
  shelfBoard: {
    height: 16,
    backgroundColor: "#E2E8DF",
    borderRadius: 8,
    borderBottomWidth: 4,
    borderBottomColor: "#C9D4C4",
    marginHorizontal: 4,
    shadowColor: "#5C6570",
    shadowOpacity: 0.15,
    shadowRadius: 3,
    shadowOffset: { width: 0, height: 2 },
    elevation: 2,
    zIndex: 1,
    marginBottom: 10 },
`;
    content = content.substring(0, styleStart) + newStyles + content.substring(styleEnd);
    console.log("Replaced Styles!");
} else {
    console.log("Could not find Styles boundaries");
}

fs.writeFileSync('mobile-app/app/journal.tsx', content);

File: do_shelf_update.py
import re

with open('mobile-app/app/journal.tsx', 'r', encoding='utf-8') as f:
    content = f.read()

# Replace the bottom section
bottom_pattern = re.compile(r'<View style={\[styles.bottomSection.*?</View>\s*</View>', re.DOTALL)

new_bottom = """<View style={[styles.bottomSection, compact && styles.bottomSectionCompact]}>
          <View style={styles.shelfHeaderRow}>
            <Text style={[styles.shelfTitle, compact && styles.shelfTitleCompact]}>Your Journals</Text>
            <Pressable style={styles.viewAllButton} onPress={() => router.push("/journal-entries")}>
              <Text style={styles.viewAllText}>View All</Text>
              <Ionicons name="chevron-forward" size={16} color="#6B7B8C" />
            </Pressable>
          </View>

          <View style={styles.cabinetBackground}>
            <View style={styles.cabinetInnerShadow} />
            <View style={styles.booksRow}>
              {/* Book 1: Muni Mode */}
              <Pressable style={[styles.shelfBookItem, compact && styles.shelfBookItemCompact]} onPress={() => router.push("/write-entry?mode=new-muni")}>
                <Image source={BOOK_IMAGE} style={styles.bookBaseImage} resizeMode="stretch" />
                <View style={styles.bookCoverContent}>
                  <Text style={styles.bookCoverTitle}>Guided{"\n"}Journal</Text>
                  <Text style={styles.bookCoverSubtitle}>w/ Muni</Text>
                </View>
                <View style={[styles.bookDeco, styles.bookDecoMuni]}>
                  <Ionicons name="chatbubbles" size={18} color="#FFFFFF" />
                </View>
              </Pressable>

              {/* Book 2: Solo Mode */}
              <Pressable style={[styles.shelfBookItem, compact && styles.shelfBookItemCompact]} onPress={() => router.push("/write-entry?mode=new-solo")}>
                <Image source={BOOK_IMAGE} style={styles.bookBaseImage} resizeMode="stretch" />
                <View style={styles.bookCoverContent}>
                  <Text style={styles.bookCoverTitle}>Solo{"\n"}Journal</Text>
                  <Text style={styles.bookCoverSubtitle}>Free Write</Text>
                </View>
                <View style={[styles.bookDeco, styles.bookDecoSolo]}>
                  <Ionicons name="pencil" size={18} color="#FFFFFF" />
                </View>
              </Pressable>
            </View>

            {/* Shelf Board */}
            <View style={styles.shelfBoardTop} />
            <View style={styles.shelfBoardFront}>
               <View style={styles.shelfWoodGrain} />
            </View>
            <View style={styles.shelfUnderShadow} />
          </View>
        </View>"""

content = bottom_pattern.sub(new_bottom, content, count=1)

# Replace Styles
style_start_idx = content.find('  shelfHeaderRow: {')
style_end_idx = content.find('  modalBackdrop: {')

if style_start_idx != -1 and style_end_idx != -1:
    new_styles = """  shelfHeaderRow: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between",
    marginBottom: 12,
    paddingHorizontal: 4 },
  shelfTitle: {
    color: "#34475A",
    fontSize: 20,
    lineHeight: 26,
    fontFamily: "Outfit-Bold" },
  shelfTitleCompact: {
    fontSize: 18,
    lineHeight: 24 },
  viewAllButton: {
    flexDirection: "row",
    alignItems: "center",
    columnGap: 2 },
  viewAllText: {
    color: "#6B7B8C",
    fontSize: 14,
    lineHeight: 18,
    fontFamily: "Outfit-Bold" },
  cabinetBackground: {
    backgroundColor: "#CBA471",
    borderRadius: 16,
    borderWidth: 3,
    borderColor: "#A2794A",
    overflow: "hidden",
    position: "relative",
    marginHorizontal: 2,
    shadowColor: "#000",
    shadowOpacity: 0.15,
    shadowRadius: 8,
    shadowOffset: { width: 0, height: 4 },
    elevation: 4 },
  cabinetInnerShadow: {
    position: "absolute",
    top: 0,
    left: 0,
    right: 0,
    height: 40,
    backgroundColor: "rgba(0,0,0,0.15)",
    zIndex: 0 },
  booksRow: {
    flexDirection: "row",
    justifyContent: "space-around",
    alignItems: "flex-end",
    paddingHorizontal: 20,
    paddingTop: 30,
    paddingBottom: 0,
    zIndex: 2 },
  shelfBookItem: {
    width: 110,
    height: 154,
    position: "relative",
    shadowColor: "#000",
    shadowOpacity: 0.3,
    shadowRadius: 6,
    shadowOffset: { width: 4, height: 2 },
    elevation: 5 },
  shelfBookItemCompact: {
    width: 95,
    height: 133 },
  bookBaseImage: {
    width: "100%",
    height: "100%",
    borderRadius: 4 },
  bookCoverContent: {
    position: "absolute",
    top: 24,
    left: 0,
    right: 0,
    alignItems: "center",
    paddingHorizontal: 12 },
  bookCoverTitle: {
    color: "#3D4E3A",
    fontFamily: "Outfit-Bold",
    fontSize: 15,
    textAlign: "center",
    lineHeight: 18,
    marginBottom: 2 },
  bookCoverSubtitle: {
    color: "#5C7356",
    fontSize: 11,
    fontFamily: "Outfit-Bold",
    textAlign: "center" },
  bookDeco: {
    position: "absolute",
    width: 34,
    height: 34,
    borderRadius: 17,
    alignItems: "center",
    justifyContent: "center",
    borderWidth: 2,
    borderColor: "#FFFFFF",
    shadowColor: "#000",
    shadowOpacity: 0.2,
    shadowRadius: 4,
    shadowOffset: { width: 0, height: 2 },
    elevation: 3 },
  bookDecoMuni: {
    backgroundColor: "#7BCB45",
    bottom: 12,
    right: -8 },
  bookDecoSolo: {
    backgroundColor: "#FFA726",
    bottom: 12,
    right: -8 },
  shelfBoardTop: {
    height: 10,
    backgroundColor: "#E4C59D",
    zIndex: 1 },
  shelfBoardFront: {
    height: 20,
    backgroundColor: "#B88645",
    zIndex: 1,
    overflow: "hidden" },
  shelfWoodGrain: {
    backgroundColor: "rgba(0,0,0,0.05)",
    height: 2,
    width: "100%",
    marginTop: 4 },
  shelfUnderShadow: {
    height: 24,
    backgroundColor: "rgba(0, 0, 0, 0.25)" },
"""
    content = content[:style_start_idx] + new_styles + content[style_end_idx:]
    
with open('mobile-app/app/journal.tsx', 'w', encoding='utf-8') as f:
    f.write(content)
print("Updated successfully via python!")

